### PR TITLE
Add `chunk` function for splitting an array into equal-sized chunks

### PR DESCRIPTION
**Pull request by jdidion**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Valid examples WDL's were added or updated to the SPEC.md (see the [guide](https://github.com/openwdl/wdl-tests/blob/main/docs/MarkdownTests.md) on writing markdown tests)

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -9481,6 +9482,72 @@ Example output:
...
+
+```wdl
+version 1.2
+workflow chunk_array {
+  Array[String] s1 = ["a", "b", "c", "d", "e", "f"]
+  Array[String] s2 = ["a", "b", "c", "d", "e"]
+  Array[String] s3 = ["a", "b"]
+  Array[String] s4 = []
+  
+  scatter (a in split(s1, 3)) {
```
Comment (markjschreiber) [inline SPEC.md]: Not sure if this scatter block is needed to demonstrate the chunk function?

**Review by markjschreiber — CHANGES_REQUESTED** [review]
Is the scatter at line 9514 required?

File: SPEC.md (hunk 1 above)
Comment (wleepang) [inline SPEC.md]: should this be:
```
scatter (a in chunk(s1, 3)) { ...
```

Comment (jdidion) [reply SPEC.md]: yes thanks

Comment (jdidion) [reply SPEC.md]: I'm open to suggestions if you think another example would be better.

**Review by markjschreiber — APPROVED** [review]

File: SPEC.md [hunk 2]
```diff
@@ -9592,6 +9593,72 @@ Example output:
...
 
+### ✨ `chunk`
+
+```
+Array[Array[X]] chunk(Array[X], Int)
+```
+
+Given an array and a length *n*, splits the array into consecutive, non-overlapping arrays of *n* elements. If the length of the array is not a multiple *n* then the final sub-array will have `length(array) % n` elements.
+
+This function is the inverse of [`flatten`](#flatten); i.e., `flatten`ing a `chunk` array will always yield the original array.
```
Comment (illusional) [inline SPEC.md]: I'm meh on this line, I don't know if it adds much.

File: SPEC.md [hunk 3]
```diff
@@ -9592,6 +9593,72 @@ Example output:
...
+
+**Parameters**
+
+1. `Array[X]`: The array to split. May be empty.
+2. `Int`: The desired length of the sub-arrays. Must be > 0.
+
+**Returns**: An array of sub-arrays, where each sub-array is the same length except possibly the last one.
```
Comment (illusional) [inline SPEC.md]:
```suggestion
**Returns**: An array of sub-arrays, where each sub-array is of length `N` except possibly the last one.
```

**Review by patmagee — COMMENTED** [review]
@jdidion what is the behaviour if the number of requested chunks is greater then the length of the array? An error?

**Comment by jdidion** [comment]
@patmagee the parameter is the requested chunk size, not the number of chunks. And it is stated that the last chunk can be shorter than the requested size. So e.g. if the array is of length 1 and I request a chunk size of 5, then the returned value will be an array of one array that contains one element.

File: SPEC.md (hunk 2 above)
Comment (jdidion) [reply SPEC.md]: Removed

**Pull request merged**